sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -1187,7 +1187,7 @@ class ForStatementExec(
       OneRowRelation()
     )
     val setIdentifierToCurrentRow =
-      SetVariable(Seq(UnresolvedAttribute(varName)), projectNamedStruct)
```

Comment:
This is not the first time we made similar mistakes. We should probably remove `UnresolvedAttribute.apply` which takes a string and parses it.